pkg/virt-handler/virtwrap/manager.go

```diff
@@ -312,20 +315,37 @@ func NewLibvirtDomainManager(connection Connection, recorder record.EventRecorde
 	return &manager, nil
 }
 
+func aggregateMappingErrors(mappingErrs []error) error {
+	log := logging.DefaultLogger()
+	if len(mappingErrs) == 0 {
+		return nil
+	}
+	if len(mappingErrs) == 1 {
+		log.Error().Msgf("Error mapping domain spec: %v", mappingErrs[0])
+		return mappingErrs[0]
+	} else {
+		errorStrings := []string{}
+		for _, err := range mappingErrs {
+			errorStrings = append(errorStrings, err.Error())
+		}
+		log.Error().Msgf("Multiple errors mapping domain spec: %s", strings.Join(errorStrings, ", "))
+		return fmt.Errorf("multiple errors mapping domain spec: %s", strings.Join(errorStrings, ", "))
+	}
+}
+
 func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) error {
 	var wantedSpec api.DomainSpec
 	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
 	if len(mappingErrs) > 0 {
-		// TODO: proper aggregation
-		return mappingErrs[0]
+		return aggregateMappingErrors(mappingErrs)
```

Comment:
Could you use `errors.Aggregate` from client-go instead? It does the same thing already